src/Avalonia.Layout/Layoutable.cs

```diff
@@ -693,14 +692,37 @@ protected virtual Size ArrangeOverride(Size finalSize)
             return finalSize;
         }
 
-        /// <inheritdoc/>
-        protected override sealed void OnVisualParentChanged(IVisual oldParent, IVisual newParent)
+        /// <summary>
+        /// Invalidates measure for this instance and all visual children.
+        /// </summary>
+        protected void InvalidateSelfAndDescendantsMeasure()
```

Comment:
Hmm, making this a protected method makes it an API, which feel kinda wrong? It's only for use in very particular scenarios, and shouldn't really be used in any old derived control.

I'm not sure what the best alternative is though, given that it's needed in 2 different Avalonia assemblies.

Reply:
It might be time to start generating reference assemblies and strip everything with `[PrivateAPI]` attribute.

Reply:
Moved it to `LayoutHelper` for now.